### PR TITLE
rails authentications: Members only

### PR DESCRIPTION
**Pull request by mahmoud717**
in this project I created a rails app that does the following:
1- added authentication for users.
2- when a user is logged in he can create a post and see the other posts with the poster name.
3- if the user is not logged in he can only see the posts without any names.

---------------------------------------------------------------------------------------------------------------------------------------

there are some issues with the commit history that I had to neglect because I created the repo from a template that had to branches( master & development ) and worked on the development branch directly, because of that I can't merge them together because they have no common commit history.

i will take care of that the next time..

### CODE REVIEW

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Request changes :recycle:
### Good job :+1: However, you need to fix some issues and implement some missing requirements as mentioned in my comments :point_down:
**_(I would have added some of the following comment in the affected section of the commit history but it not available, due to a short commit history)_**

- **_Fix the configuration of the database it throws an error when I tried to run the migration command_**

- **_Remove all the logic from the views :point_down:_**

  * [ ] There should be NO application logic in the views (no `Post.new`, `Comment.create`, `current_user.friends.find_by(...)`...). Everything connected with data processing should be encapsulated in models and used in controllers.
  * [ ] View-related logic can be encapsulated in View helpers (app/helpers/\*_helper.rb)
  * [ ] There should be absolutely NO logic in views.

- **_Use `before_action` to restrict access to the `:new` and `:create` methods_**

#### _Reviewed by Temitayo Adeleke_

**Comment by mahmoud717** [comment]
- there are no errors happening when I try to run the migrations, not even on heruko.
- the only logic inside the views is using if statements and helpers predefined by the devise gem (user_signed_in?).

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <div class="container">
-    <%if @posts.count == 0 && !user_signed_in? %>
+    <%if @posts.is_empty? && !user_signed_in? %>
     <h3 class="mt-5 text-center"> There are no posts right now!!</h3>
     <%end %>
```
Comment (maelfosso) [inline app/views/posts/index.html.erb]: Remove logic from views.
Implement them in views helpers

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,29 +1,26 @@
...
 
   # GET /posts
   # GET /posts.json
   def index
     @posts = Post.all
-    if user_signed_in?
-    @post = current_user.posts.build
-    end
+    @post = current_user.posts.build if user_signed_in?
     @users = User.all
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb:10]: Why this variables?
You don't use it in the views, kindly remove it.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for making the suggested changes.
However, there are still some changes to do

1. Please, remove logic from views. Implement into view helpers

2. I got this error when launching your app
Screenshot at 2020-10-09 22:48:15

Also, this one
Screenshot at 2020-10-09 22:49:24
Generally, helpers are defined for each controller

Kindly fix those bugs and submit a new code review request
Thanks

Mael FOSSO

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (mahmoud717) [reply app/controllers/posts_controller.rb:10]: because I'm creating posts on the index page so I have to have a post created to not get an error

Comment (mahmoud717) [reply app/controllers/posts_controller.rb:10]: and the one in the new action isn't invoked unless I click create.
if I don't do that I'll get an error when I try to render the index page.

**Pull request closed without merging**